task_4/andrii.lukianchykov/task_4/src/main/java/com/bookshop/bookdatabase/BookDataBase.java

```diff
@@ -0,0 +1,159 @@
+package com.bookshop.bookdatabase;
+
+import static java.lang.System.out;
+
+import com.bookshop.book.Book;
+import com.bookshop.helpTests.IncorrectSymbol;
+import com.bookshop.helpTests.ListIsEmptyException;
+import com.bookshop.helpTests.ListIsNotEmptyException;
+import com.bookshop.helpTests.ListIsNullException;
+import com.bookshop.response.Response;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class BookDataBase {
+
+    private List<Book> listOfBooks = new ArrayList<>();
+    private Book book = new Book();
+    private Response response = new Response();
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("List of Books: ").append(listOfBooks);
+        return sb.toString();
+    }
+    public void showBookList(){
+        out.println("List of books: " + listOfBooks.toString());
+    }
+
+    public void showBooksServletOutput() {
+
+        if( listOfBooks.size() == 0) {
+            throw new ListIsEmptyException("List is empty");
+        }
+        if(listOfBooks.isEmpty()) {
+            for(Book b : listOfBooks){
+                response.getMiddle(b.getTitleBook(),b.getAuthorBook(),b.getYearBook());
+                showBookList();
+            }
+        }else{
+            response.getOutPutElse();
+        }
+    }
+
+    public void defaultValuesListOfBooks() {
+
+        if (listOfBooks.size() != 0) {
+
+            listOfBooks.add(new Book("Legend of Searcher", "Terry Goodkind", "2008"));
+            listOfBooks.add(new Book("Comics", "Overwatch", "2017"));
+            listOfBooks.add(new Book("Limited Collection of how to code right", "Java Enetrprise", "2017"));
+
+        } else{
+            throw new ListIsNotEmptyException("List is not empty");
+        }
+    }
+
+    public List<Book> findBookByAuthor(String author) {
+
+        List<Book> books = new ArrayList<Book>();
+
+        if( listOfBooks.size() == 0) {
+            throw new ListIsEmptyException("List is empty");
+        }
+
+        if (listOfBooks.isEmpty()) {
+            for (Book b : listOfBooks) {
+                if (b.getAuthorBook().equals(author)) {
+                    response.getMiddle(b.getTitleBook(),b.getAuthorBook(),b.getYearBook());
+                    books.add(book);
+                }
+            }
+        }else{
+            response.getSearchElse();
+        }
+
+        return books;
+    }
+
+    public void addNewBook() {
```

Comment:
This method looks weird. You're checking that the lisf of books is empty and then throwing an exception. What is the purpose of it?